### PR TITLE
setup.py: Fix vim-vint installation bug

### PR DESCRIPTION
**Pull request by Madhukar-SH**
Fixed vim-vint setup command which was causing issues
in systems having python version older than 3.4

Fixes https://github.com/coala/coala/issues/3878

### CODE REVIEW

**Comment by Madhukar-SH** [comment]
Sorry coala is broken on my system for some reason, hence not able to run it and check for code errors.

File: setup.py [hunk 1]
```diff
@@ -55,7 +66,6 @@ def run(self):
 required += [req for req in bear_required
              if not req.startswith('language-check')]
 
-
```
Comment (meetmangukiya) [inline setup.py]: unrelated change

Comment (meetmangukiya) [inline setup.py]: You are not supposed to make this change

Comment (Madhukar-SH) [inline setup.py]: Removing empty line?

Comment (Madhukar-SH) [inline setup.py]: Undo-ing it

File: setup.py [hunk 2]
```diff
@@ -43,6 +43,17 @@ def run(self):
 with open('bear-requirements.txt') as requirements:
     bear_required = requirements.read().splitlines()
 
+version = sys.version_info  # Checks if python version is <3.4
+# If yes changes the vim-vint~=x.x to vim-vint=x.x
+if(version[0] == 3 and version[1] <= 4):
+    vim_pkg = 'vim-vint'
+    vim_package_line = [line for line in bear_required if vim_pkg in line]
```
Comment (Techievena) [inline setup.py]: Why the braces?

Comment (Techievena) [inline setup.py]: `line for line in bear_required if vim_pkg in line` Please split this

Comment (Madhukar-SH) [inline setup.py]: Braces because its a list comprehension https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions

Comment (Madhukar-SH) [inline setup.py]: Is it really necessary to split it and write it as `for loop with if inside` ?

Comment (Madhukar-SH) [inline setup.py]: List comprehension is a bit faster actually

Comment (Techievena) [inline setup.py]: Well didn't knew about such a thing. Then I think its fine. :+1:

File: setup.py [hunk 3]
```diff
@@ -43,6 +43,17 @@ def run(self):
 with open('bear-requirements.txt') as requirements:
     bear_required = requirements.read().splitlines()
 
+version = sys.version_info  # Checks if python version is <3.4
+# If yes changes the vim-vint~=x.x to vim-vint=x.x
+if(version[0] == 3 and version[1] <= 4):
```
Comment (jayvdb) [inline setup.py]: See https://github.com/coala/coala/issues/3878#issuecomment-288269752 .
This doesnt include when the version is 2.x, which is where the problem is.

**Comment by jayvdb** [comment]
Closing as the bug hasnt been reproduced yet.
Cant fix what you cant see.

File: setup.py (hunk 3 above)
Comment (Madhukar-SH) [inline setup.py]: I made this since we never use python 2.x for installing coala-bears

**Comment by Madhukar-SH** [comment]
As the issue post says, this was problem with specific versions of Linux mint 17.x where newer versions of python wasn't available

**Comment by Madhukar-SH** [comment]
I felt this was pretty much it https://www.bountysource.com/issues/38134858-177-breaks-pip-install-for-python2
Linux mint 17.3 64 bit

Check the comment of dezza (27days ago from today)

**Comment by jayvdb** [comment]
Please continue the analysis discussion on https://github.com/coala/coala/issues/3878 , not here.
